pkg/config/node/minimum_kubelet_version.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver/v4"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// An error to be returned when kubelet version is lower than specified minimum kubelet version.
+// Used to differentiate between a parsing failure, and a failure because the kubelet is out of date.
+var ErrKubeletOutdated = fmt.Errorf("kubelet version is outdated")
+
+// ValidateMinimumKubeletVersion takes a list of nodes and a currently set min version.
+// It parses the min version and iterates through the nodes, comparing the version of the kubelets
+// to the min version.
+// It will error if any nodes are older than the min version.
+func ValidateMinimumKubeletVersion(nodes []*corev1.Node, minimumKubeletVersion string) error {
```

Comment:
I think this function should be part of an admission plugin defined in the server. If so, could you explain why it was added to the library-go?

Reply:
this library is to share code between [KAS](https://github.com/openshift/kubernetes/pull/2104) and [hypershift](https://github.com/openshift/hypershift/pull/4980) it seemed cleanest to me to put these library functions here, even though not all of them are needed by both